package dataobj;

message Message {
	required uint32          obj_no = 1;
	required uint32          offset = 2;
	required uint32          length = 3;
	enum trace_operator_t {
		operator_read = 1;
		operator_write = 2;
		operator_trim = 3;
	}
	required trace_operator_t operator_t = 4;	
	required float             timestamp = 5;
	required int32          ec_index = 6;
	enum request_type_t {
		need_flash_info = 1;
		not_need_flash_info = 2;
		shut_cluster = 3;
	}
	required request_type_t rq_type = 7; 
	required float             flash_utilization = 8;
	required float             flash_victim_utilization = 9;
	required float             flash_full_blk_utilization = 10;
	required uint32          node_nr_erases = 11;
	required float             local_log_utilization = 12;
}
